.github/actions/nf-test-shard/action.yml

```
@@ -0,0 +1,38 @@
+name: "Get number of shards"
+description: "Get the number of nf-test shards for the current CI job"
+outputs:
+  shards:
+    description: "Array of shard numbers"
+    value: ${{ steps.shards.outputs.shards }}
+  total_shards:
+    description: "Total number of shards"
+    value: ${{ steps.shards.outputs.total_shards }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Install nf-test
+      uses: nf-core/setup-nf-test@v1
+      with:
+        version: ${{ env.NFT_VER }}
+
+    - name: Get number of shards
+      id: shards
+      shell: bash
+      run: |
+        nftest_output=$(nf-test test --dry-run --changed-since HEAD^ --filter pipeline)
+        echo "nftest_output: $nftest_output"
+        if echo "$nftest_output" | grep -q 'Found 0 related test(s)'; then
+          shards="[]"
+          total_shards=0
+        else
+          number_of_shards=$(echo "$nftest_output" | grep -o 'Found [0-9]* related test' | tail -1 | awk '{print $2}')
+          shards=$(seq 1 "$number_of_shards" | jq -R . | jq -s .)
```

```suggestion
          shards=$(seq 1 "$number_of_shards" | jq -R . | jq  -c -s .)
```
Github doesn't like multiline strings as output